lib/puppet/provider/registry_value/registry.rb

```diff
@@ -61,8 +63,15 @@ def flush
 
   def destroy
     Puppet.debug("Destroying registry value: #{self}")
+    # On Ruby 2.1.x, due to https://bugs.ruby-lang.org/issues/10820, we see
+    # a FileNotFound error - hence an FFI re-implementation inside destroy
     hive.open(subkey, Win32::Registry::KEY_ALL_ACCESS | access) do |reg|
-      reg.delete_value(valuename)
+      from_string_to_wide_string(valuename) do |valuename_ptr|
+        if RegDeleteValueW(reg.hkey, valuename_ptr) != 0
+          msg = "Failed to delete registry value #{valuename} at #{reg.keyname}"
+          raise Puppet::Util::Windows::Error.new(msg)
```

Comment:
Does this move forward the version compatibility? 3.3.0 https://github.com/puppetlabs/puppetlabs-registry/blob/master/metadata.json#L32